File: app/src/main/java/com/anokmik/tripassistant/trip/list/TripsListPresenter.java
package com.anokmik.tripassistant.trip.list;

import com.anokmik.persistence.model.Trip;
import com.anokmik.persistence.repository.TripRepository;
import com.anokmik.tripassistant.databinding.adapter.OnItemClickListener;
import com.anokmik.tripassistant.databinding.adapter.ViewHolderPresenter;

import java.util.List;

public final class TripsListPresenter implements TripsListContract.Presenter, OnItemClickListener<Trip> {

    private final TripsListContract.View view;
    private final TripRepository tripRepository;
    private final ViewHolderPresenter<Trip> tripViewHolderPresenter;

    public TripsListPresenter(TripsListContract.View view) {
        this.view = view;
        this.tripRepository = new TripRepository();
        this.tripViewHolderPresenter = buildRowPresenter(view);
    }

    @Override
    public List<Trip> getTrips() {
        return tripRepository.getAll();
    }

    @Override
    public ViewHolderPresenter<Trip> getViewHolderPresenter() {
        return tripViewHolderPresenter;
    }

    @Override
    public void onItemClick(Trip item, int position) {
        view.showTrip(item);
    }

    private ViewHolderPresenter<Trip> buildRowPresenter(TripsListContract.View view) {
        return new ViewHolderPresenter.Builder<Trip>(view.getRowItemLayoutId(), view.getItemBindingId()).setItemClickListener(this).build();
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/trip/list/TripsListPresenter.kt
package com.anokmik.tripassistant.trip.list

import com.anokmik.persistence.model.Trip
import com.anokmik.persistence.repository.TripRepository
import com.anokmik.tripassistant.databinding.adapter.OnItemClickListener
import com.anokmik.tripassistant.databinding.adapter.ViewHolderPresenter

class TripsListPresenter(private val view: TripsListContract.View) : TripsListContract.Presenter, OnItemClickListener<Trip> {

    private val tripRepository = TripRepository()

    override val trips = tripRepository.getAll()

    override val viewHolderPresenter = ViewHolderPresenter.Builder<Trip>(view.rowItemLayoutId, view.itemBindingId).setItemClickListener(this).build()

    override fun onItemClick(item: Trip?, position: Int) = view.showTrip(item)

}